topi/python/topi/arm_cpu/conv2d.py

```diff
@@ -340,8 +343,8 @@ def _decl_winograd(cfg, data, kernel, strides, padding, dilation, layout, out_dt
     K = CO
     C = CI
 
-    H = (IH + 2 * HPAD - 3) // HSTR + 1
-    W = (IW + 2 * WPAD - 3) // WSTR + 1
+    H = (IH + 2 * HPAD - KH) // HSTR + 1
```

Comment:
Could you help to change `HPAD` / `WPAD` to `pad_top + pad_down` / `pad_left + pad_right` like `spatial_pack` do? We could get `pad_top / down / left / right` from `get_pad_tuple`. Thanks.

Reply:
@FrozenGene ,
It is done. Please help double check for mistakes, i did it in quite many places, and altered more than that in cuda's winograd (added ```dilation``` related assertion).

Reply:
Could you help to explain why we add `dilation` assertion in CUDA? For CUDA, better way maybe should like we do in ARM CPU / Mali / nn.conv2d_nchw (that is another story), not just `dialte` kernel directly.

Reply:
@FrozenGene ,

 Removed the inserted```assert``` . Still concerned if is good that ```HPAD = WPAD = 1``` is removed from ```pre-transformed``` case handler.
 
Can tell if padding related lines are fine now for cuda ?

Reply:
@FrozenGene ,

 Ok, not touching ```dilation``` related anymore, maybe will subject for another PR. I reverted this part from cuda/winograd . Let's remain only at ```tile_size``` introduction for ```cuda``` .
 Please confirm / reslove-conversation if find no problems.

Reply:
Yes. Thanks. We could do `dilation` and `padding` in another PR. Make sense.